pygmt/src/grdcut.py

```diff
@@ -99,13 +98,23 @@ def grdcut(grid, **kwargs):
     >>> # 12° E to 15° E and a latitude range of 21° N to 24° N
     >>> new_grid = pygmt.grdcut(grid=grid, region=[12, 15, 21, 24])
     """
-    with GMTTempFile(suffix=".nc") as tmpfile:
-        with Session() as lib:
-            with lib.virtualfile_in(check_kind="raster", data=grid) as vingrd:
-                if (outgrid := kwargs.get("G")) is None:
-                    kwargs["G"] = outgrid = tmpfile.name  # output to tmpfile
-                lib.call_module(
-                    module="grdcut", args=build_arg_list(kwargs, infile=vingrd)
-                )
+    inkind = data_kind(grid)
+    match inkind:
+        case "image" | "grid":
+            outkind = inkind
+        case "file":
+            realpath = which(grid, download="a")
+            if isinstance(realpath, list):
+                realpath = realpath[0]
+            outkind = "image" if realpath.endswith(".tif") else "grid"
```

Comment:
When a file is given, there are two different ways to determine if the file is a grid or an image:

1. Based on file extensions
2. Read the file (header only if possible) via `xarray` or `rasterio` and check the data dims.

I guess option 2 is more robust and preferred.

Reply:
Currently use option 1 because it's easy for coding.